### PR TITLE
Split Mazda cars to : CX5, CX9, and Mazda3

### PR DESCRIPTION
**Pull request by Jafaral**
All Mazda cars were registered as CX-5. This PR spilts them to different models 
and fills in the relevant params and allows PID tuning per model.

Signed-off-by: Jafar Al-Gharaibeh <to.jafar@gmail.com>

### CODE REVIEW

File: selfdrive/car/mazda/values.py [hunk 1]
```diff
@@ -41,27 +44,43 @@ class Buttons:
...
+    }
+  ],
 
+  CAR.CX9: [
     # CX-9 2017 Australia - old CAM connector
     {
       64: 8, 70: 8, 80: 8, 117: 8, 118: 8, 120: 8, 121: 8, 130: 8, 134: 8, 138: 8, 145: 8, 154: 8, 155: 8, 157: 8, 158: 8, 159: 8, 253: 8, 304: 8, 305: 8, 357: 8, 358: 8, 359: 8, 512: 8, 514: 8, 515: 8, 522: 8, 529: 8, 533: 8, 535: 8, 539: 8, 540: 8, 541: 8, 542: 8, 543: 8, 552: 8, 576: 8, 577: 8, 578: 8, 579: 8, 580: 8, 581: 8, 582: 8, 583: 8, 605: 8, 606: 8, 628: 8, 832: 8, 976: 8, 977: 8, 978: 8, 1034: 8, 1045: 8, 1056: 8, 1061: 8, 1067: 8, 1078: 8, 1085: 8, 1086: 8, 1088: 8, 1093: 8, 1108: 8, 1114: 8, 1115: 8, 1116: 8, 1139: 8, 1143: 8, 1147: 8, 1154: 8, 1157: 8, 1160: 8, 1163: 8, 1166: 8, 1170: 8, 1177: 8, 1180: 8, 1183: 8, 1233: 8, 1236: 8, 1237: 8, 1238: 8, 1239: 8, 1241: 8, 1242: 8, 1243: 8, 1244: 8, 1247: 8, 1264: 8, 1266: 8, 1267: 8, 1269: 8, 1271: 8, 1272: 8, 1274: 8, 1277: 8, 1278: 8, 1409: 8, 1416: 8, 1425: 8, 1430: 8, 1435: 8, 1440: 8, 1446: 8, 1456: 8, 1479: 8
     },
 
+    # CX-9 2018
```
Comment (adeebshihadeh) [inline selfdrive/car/mazda/values.py:56]: Why are these commented out?

Comment (Jafaral) [reply selfdrive/car/mazda/values.py:56]: They collide with CX-5 2019 FP. I left them there for CX-9 end users so they can put them back end and drop the CX-5 FP. I can drop them if needed.

Comment (adeebshihadeh) [reply selfdrive/car/mazda/values.py:56]: yes, let's remove them before merging

Comment (Jafaral) [reply selfdrive/car/mazda/values.py]: Ok, dropped them. I'm working on FP2, hopefully this problem will be gone before too long

**Comment by adeebshihadeh** [comment]
LGTM, will merge once we can get routes for each of the new models and tests pass.

**Comment by Jafaral** [comment]
BTW @adeebshihadeh, I  put all of the new models  under a GEN1 set for  Mazda generation 1 ADAS hardware. This  make it easy to separate the logic for GEN2 hardware when we add support for it.

**Comment by adeebshihadeh** [comment]
Do the gen 2 models have the steering lockout or more torque?

**Comment by Jafaral** [comment]
Hard to tell without having a car myself, but users reported no lockout at least, but maybe more torque as well.

**Comment by Jafaral** [comment]
@adeebshihadeh  the new routes are added,  please see https://github.com/commaai/openpilot/pull/2097/commits/b025451e7a75d070da1061b249314a63ad74d865

File: selfdrive/test/test_car_models.py [hunk 2]
```diff
@@ -433,6 +433,18 @@ def get_route_log(route_name):
     'carFingerprint': MAZDA.CX5,
     'enableCamera': True,
   },
+  "10b5a4b380434151|2020-08-26--17-11-45": {
+    'carFingerprint': MAZDA.CX9,
+    'enableCamera': True,
+  },
+  "74f1038827005090|2020-08-27--10-23-22": {
```
Comment (adeebshihadeh) [inline selfdrive/test/test_car_models.py:440]: this route isn't uploaded

File: selfdrive/test/test_car_models.py [hunk 3]
```diff
@@ -433,6 +433,18 @@ def get_route_log(route_name):
...
+    'carFingerprint': MAZDA.Mazda3,
+    'enableCamera': True,
+  },
+  "0f702597446474bf|2020-08-12--08-35-04": {
```
Comment (adeebshihadeh) [inline selfdrive/test/test_car_models.py:444]: neither is this one

Comment (Jafaral) [reply selfdrive/test/test_car_models.py:444]: How do I know if they are uploaded? These routes show up on useradmin page.

Comment (adeebshihadeh) [reply selfdrive/test/test_car_models.py:444]: all the rlogs and fcamera need to be uploaded too

**Comment by Jafaral** [comment]
@adeebshihadeh I updated the route for Mazda 6, can you please check if all of the routes are now passing ?

File: selfdrive/test/test_car_models.py [hunk 4]
```diff
@@ -433,6 +433,18 @@ def get_route_log(route_name):
...
   },
+  "10b5a4b380434151|2020-08-26--17-11-45": {
+    'carFingerprint': MAZDA.CX9,
+    'enableCamera': True,
+  },
+  "74f1038827005090|2020-08-26--20-05-50": {
+    'carFingerprint': MAZDA.Mazda3,
+    'enableCamera': True,
+  },
+  "0f702597446474bf|2020-09-01--16-53-30": {
```
Comment (adeebshihadeh) [inline selfdrive/test/test_car_models.py:444]: this route isn't uploaded

File: selfdrive/car/mazda/interface.py [hunk 5]
```diff
@@ -33,18 +33,35 @@ def get_params(candidate, fingerprint=gen_empty_fingerprint(), has_relay=False,
...
       ret.lateralTuning.pid.kf = 0.00006
+    elif candidate == CAR.Mazda3:
+      ret.mass = 2875 * CV.LB_TO_KG + STD_CARGO_KG
+      ret.wheelbase = 2.7
+      ret.steerRatio = 14.0
+      ret.lateralTuning.pid.kiBP, ret.lateralTuning.pid.kpBP = [[0.], [0.]]
+      ret.lateralTuning.pid.kpV, ret.lateralTuning.pid.kiV = [[0.19], [0.019]]
+      ret.lateralTuning.pid.kf = 0.00006
+
+      #Mazda6 params
```
Comment (adeebshihadeh) [inline selfdrive/car/mazda/interface.py:58]: remove these comments

**Comment by adeebshihadeh** [comment]
Merging, test failure is unrelated to the PR

**Pull request merged**